architecture/overview.md

```diff
@@ -2,50 +2,54 @@
 
-- Margin Engine
-- Virtual automated market maker (vAMM)
-- Insurance Fund
-- Fee Pool
-- Pricefeed
+1. Users are able to interact with all public functions, including open/close positions and liquidations
+2. Governance is able to perform config updates and contract migrations for both the insurance fund and margin engine contracts
+3. A portion of trading fees are deposited into the insurance fund
+4. Funds may be withdrawn to cover a shortfall
+5. In an emergency governance can trigger shutdown of all vAMMs through the insurance fund
+6. The engine deposits a portion of the fees into the fee pool
+7. Revenue generated is redistributed to token holders
+8. The vAMM manages protocol liquidity during all input and output swaps triggered by the margin engine
+9. Index price is retrieved from a pricefeed during the settlement of funding payments
 
-The Engine is the point of interaction for users with the protocol, and most methods in the other contracts check whether the sender is the Engine. The Engine allows people to open/close positions, and to add/remove margin from existing positions
+## Component Descriptions
 
-### Component Diagram:
+### Governance
 
-![Contract Interactions](Images/contractInteractions.png)
+Protocol governance is able to configure and migrate contracts.
 
-1. A portion of fees from transactions go to the insurance fund
-2. Sometimes, the insurance fund is used to cover shortfall and bad debt that users accrue, to make all creditors whole
-3. The insurance fund has the power to shut all the vAMMs down, in the case of an emergency
-4. The engine deposits a portion of the fees into the fee pool
-5. At some point, the funds in the fee pool are paid out to token holders
-6. The vAMM calculates the input and output prices when swapping and the Engine uses these to calculate position size
-7. The positional data that the vAMM returns is then stored in the Engine. Most of the data is user input, but the vAMM calculates the position size
-8. The vAMM calculates the funding payments, and to do so it receives pricing data from the on-chain oracle, which is implemented via the Pricefeed contract
+### Users
+
+As a decentralised protocol users are free to interact with the protocol as they wish. Users have two roles:
+
+* Traders
+* Liquidators
+
+Traders are users who take perpetual positions within the protocol. Liquidators are users who monitor the margin levels of users and perform liquidations appropriately.
 
 ### Margin Engine
 
-The Engine owns all of the contracts. This means that most of the methods that exist in the other contracts will error if the sender is not the Engine.
+The margin engine manages the positions of protocol users ans is permissoned to perform actions with the vAMMs on behalf of users. Traders use the margin engine contract to open and close positions. Additionally traders are able to deposit and remove margin from their margin account within the engine as appropriate.

```

Comment:
The margin engine manages the positions of protocol users and performs actions with the vAMMs on behalf of users. Traders use the margin engine contract to open and close positions. Additionally, traders are able to deposit and remove margin from their margin account within the engine as appropriate.